### PR TITLE
Improve Documentation: Dashboard Deployment on Toolforge and Cloud VPS

### PR DESCRIPTION
**Pull request by empty-codes**
## What this PR does
Adds Admin guide documentation explaining how the dashboard is run on Toolforge and Cloud VPS.

WikiEduDashboardTools PR link: https://github.com/WikiEducationFoundation/WikiEduDashboardTools/pull/19

This PR is an ongoing effort to improve the existing documentation as requested here: https://phabricator.wikimedia.org/T375642

### CODE REVIEW

File: docs/admin_guide.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    - [Web Server Issues](#web-server-issues)
+    - [Database Issues](#database-issues)
+    - [Data Dumps and Recovery](#data-dumps-and-recovery)
+4. [More Resources](#more-resources)
+
+## Overview
+
+The **Programs & Events Dashboard** ([outreachdashboard.wmflabs.org](https://outreachdashboard.wmflabs.org)) is a web application designed to support the global Wikimedia community in organizing various programs, including edit-a-thons, education initiatives, and other events. **[Source Code](https://github.com/WikiEducationFoundation/WikiEduDashboard/tree/wmflabs)** **[Phabricator Project](https://phabricator.wikimedia.org/project/manage/1052/)**  
+
+### Infrastructure Overview
```
Comment (ragesoss) [inline docs/admin_guide.md:20]: This should include prose describing the different elements of the infrastructure.

Comment (empty-codes) [reply docs/admin_guide.md:20]: How is this?

## Infrastructure Overview
The application is hosted on the Wikimedia Cloud VPS project [globaleducation](https://openstack-browser.toolforge.org/project/globaleducation), which provides the infrastructure for all servers, allowing the dashboard to run on virtual machines that are flexible and easily managed within Wikimedia Cloud. 

The dashboard leverages the [wikiedudashboard](https://toolsadmin.wikimedia.org/tools/id/wikiedudashboard) Toolforge tool, a collection of PHP endpoints,  to retrieve revision and article data from Wikimedia Replica databases in the Wikimedia Cloud environment. The tool is deployed at [wikiedudashboard.toolforge.org](https://wikiedudashboard.toolforge.org/) and [replica-revision-tools.wmcloud.org](https://replica-revision-tools.wmcloud.org/). **Source code: [WikiEduDashboardTools](https://github.com/WikiEducationFoundation/WikiEduDashboardTools)**


### Servers

The dashboard operates on a distributed server architecture to handle web requests, process background jobs, and store application data. Each server is dedicated to specific roles, minimizing competition for resources and improving reliability by isolating potential bottlenecks and failures.

Below is a breakdown of the key servers and their roles within the infrastructure:

1. **Web Server**  ...........................|

```  ```
This currently describes toolforge, cloud vps and the 4 web servers, are there any other aspects I should mention?

Comment (ragesoss) [reply docs/admin_guide.md:20]: This is a little off. The WikiEduDashboardTools service is currently only used at replica-revision-tools.wmcloud.org.

We do also use wikiedudashboard.toolforge.org, but only as a source of static assets for the Vega.js files for Programs & Events Dashboard (see _vega.html.haml).

Comment (ragesoss) [reply docs/admin_guide.md:20]: There is also the reference-counter API: https://toolsadmin.wikimedia.org/tools/id/reference-counter (see reference_counter_api.rb).

Comment (ragesoss) [reply docs/admin_guide.md:20]: One more: https://ruby-suspected-plagiarism.toolforge.org/ - https://toolsadmin.wikimedia.org/tools/id/ruby-suspected-plagiarism

It's probably also worth including details about the non-WMF-production services/tools that parts of the system rely on, including:
* PetScan
* PagePile
* WikiWho
* Copypatrol (the database for which is used by the ruby-suspected-plagiarism tool)

File: docs/admin_guide.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+   - **`peony-sidekiq-medium.globaleducation.eqiad1.wikimedia.cloud`**: Hosts `sidekiq-medium` for typical course updates.  
+   - **`peony-sidekiq-3.globaleducation.eqiad1.wikimedia.cloud`**: Hosts `sidekiq-long` for long-running course updates with higher queue latency.  
+
+3. **Database Server**  
+   - **`peony-database.globaleducation.eqiad1.wikimedia.cloud`**: Stores program, user, and revision data. It supports the dashboard’s data queries and updates.
+
+4. **Redis Server**  
+   - **`p-and-e-dashboard-redis.globaleducation.eqiad1.wikimedia.cloud`**: Stores all task (job) details and is shared across all Sidekiq processes for task queuing and caching.
+
+### Integrated Toolforge Tools
```
Comment (empty-codes) [inline docs/admin_guide.md:50]: @ragesoss I added all the services you mentioned plus whocolor, wikidatadiffanalyzer and liftwing api. I can also remove them if they're unnecessary though.

File: docs/admin_guide.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+     - **Capistrano** is used for deployments  
+     - Sidekiq processes hosted:  
+       - `sidekiq-default`: Manages frequently run tasks (e.g., adding courses to update queues).  
+       - `sidekiq-constant`: Handles transactional jobs (e.g., wiki edits, email notifications).  
+       - `sidekiq-daily`: Executes long-running daily update tasks.  
+
+2. **Sidekiq Servers**: These dedicated servers handle the other Sidekiq processes to isolate bottlenecks and failures:  
+   - **`peony-sidekiq.globaleducation.eqiad1.wikimedia.cloud`**: Hosts `sidekiq-short` for short-running course updates.  
+   - **`peony-sidekiq-medium.globaleducation.eqiad1.wikimedia.cloud`**: Hosts `sidekiq-medium` for typical course updates.  
+   - **`peony-sidekiq-3.globaleducation.eqiad1.wikimedia.cloud`**: Hosts `sidekiq-long` for long-running course updates with higher queue latency.
```
Comment (ragesoss) [inline docs/admin_guide.md:42]: The short and long servers are reversed. -3 is handling the short queue and the first one is handling the long queue.

File: docs/admin_guide.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+  **[[Live Tool](https://pagepile.toolforge.org/), [Source Code](https://bitbucket.org/magnusmanske/pagepile/src/master/), [Documentation](https://pagepile.toolforge.org/howto.html)]**
+
+
+### Other Integrated APIs and Third-Party Dependencies
+
+- **[PetScan](https://petscan.wmcloud.org/)**  
+    A powerful tool that can assemble lists of articles using a wide variety of data sources (including categories and templates, as well incoming and outgoing links, Wikidata relationships, and more). Users create a query on the PetScan website, which returns a PSID for that query, and that PSID is how the Dashboard connects to the PetScan API to get the list of articles. PetScan queries are dynamic; while the query for a given PSID cannot be modified, the results may change each time the the query is run, based on changes that happened on Wikipedia and Wikidata.
```
Comment (ragesoss) [inline docs/admin_guide.md:80]: These descriptions don't need to cover the tool itself. It should just focus on how the Dashboard uses each (perhaps in a little bit more detail than you have so far).

**Comment by ragesoss** [comment]
Great job!

**Pull request merged**